c/models/dt_ftrl.cc

```diff
@@ -556,16 +558,20 @@ FtrlFitOutput Ftrl<T>::fit(T(*linkfn)(T),
 
   // Training settings. By default each training iteration consists of
   // `dt_X_train->nrows` rows.
-  size_t niterations = params.nepochs;
+  size_t niterations = static_cast<size_t>(std::ceil(nepochs));
+  T frac_nepochs = nepochs - niterations + 1;
```

Comment:
if `nepochs` is integer, say `3`, then `niterations = 3` too, and then `frac_nepochs = 1`. Is this what you intended?

Reply:
Yeah, but the variable names I've used here could really cause some confusion. I've streamlined this part and added more comments.